### PR TITLE
Write documentation

### LINKED ISSUE #23: Write Documentation
Write some basic documentation about this tool.

## Task Deliverables

- [ ] Basic documentation about Steward

### PR DESCRIPTION
**Pull request by srueg**
Closes #23

### CODE REVIEW

**Comment by srueg** [comment]
@akosma can you show me an example how to properly link other documents in the SYN project? For example I want to link the Lieutenant API docs or the Steward design document.

**Comment by akosma** [comment]
> @akosma can you show me an example how to properly link other documents in the SYN project? For example I want to link the Lieutenant API docs or the Steward design document.

Of course, easy:

```
* xref:index.adoc[In-module page]
* xref:other-module:page.adoc[Page in the same component, but a different module]
** xref:topic/page.adoc[In-module page in a topic folder]
*** xref:other-module:page.adoc#fragment[A deep link to a page in the same component, but a different module]
*** xref:other-component:module:page.adoc[Page in another component]
```

[(Source)](https://docs.antora.org/antora/2.2/navigation/link-syntax-and-content/#page)

**Comment by srueg** [comment]
Thanks!
Sooo something like this would work: `xref:lieutenant-api:ROOT:index.adoc[Lieutenant API]`?

**Comment by akosma** [comment]
> Thanks!
> Sooo something like this would work: `xref:lieutenant-api:ROOT:index.adoc[Lieutenant API]`?

Yes, AFAIK it should work.

**Comment by srueg** [comment]
Ok, cool. I added some links.

File: docs/modules/ROOT/pages/index.adoc [hunk 1]
```diff
@@ -1,6 +1,45 @@
...
 = Steward
 
 image::syn.png[]
 
-This is the documentation of the Steward project.
+This is the documentation of the Steward cluster agent. Also see the xref:syn:SDDs:0009-steward-cluster-agent.adoc[SDD 0009 - Steward Cluster Agent] for more information.
+
+== Introduction
+
+The Steward cluster agent is the first part of SYN that's installed on a new cluster to manage it. It connects to the xref:lieutenant-api::index.adoc[Lieutenant API] to receive the necessary configuration and to report back the cluster state.
```
Comment (tobru) [inline docs/modules/ROOT/pages/index.adoc:9]:
```suggestion
The Steward cluster agent is the first part of Project Syn that's installed on a new cluster to manage it. It connects to the xref:lieutenant-api::index.adoc[Lieutenant API] to receive the necessary configuration and to report back the cluster state.
```

File: docs/modules/ROOT/pages/index.adoc [hunk 2]
```diff
@@ -1,6 +1,45 @@
...
+
+This is done on a minutely interval. It also checks on each run if the Argo CD components are deployed (exist) and bootstraps them if they don't exist.
```
Comment (tobru) [inline docs/modules/ROOT/pages/index.adoc:11]:
```suggestion
This is done once per minute. It also checks on each run if the Argo CD components are deployed (exist) and bootstraps them if they don't exist.
```

File: docs/modules/ROOT/pages/index.adoc [hunk 3]
```diff
@@ -1,6 +1,45 @@
...
+* Kubernetes distribution (for example `openshift4`)
+* SSH public key (used to clone the catalog git repo)
+
+
+=== Authentication
+
+In order to communicate with the API, Steward needs to authenticate to it. A bearer token is configured in the `steward` secret which is initially installed and subsequently managed by Argo CD. It's configured as an environment variable: `STEWARD_TOKEN`.
+The cluster ID of its own cluster is also configured as an environment variable: `STEWARD_CLUSTER_ID`.
+
+This API user needs permissions to `get` and `update` its own cluster object.
```
Comment (tobru) [inline docs/modules/ROOT/pages/index.adoc:35]:
```suggestion
This API user needs permissions to `get` and `update` its own Lieutenant cluster object.
```

File: docs/modules/ROOT/pages/index.adoc [hunk 4]
```diff
@@ -1,6 +1,45 @@
...
+
+
+== Bootstrapping
+
+As soon as Steward could connect to the API and got the necessary information it starts to bootstrap Argo CD. The initial setup consists of the default deployments required to run Argo CD (argocd-application-controller, argocd-redis, argocd-repo-server and argocd-server), the Argo CD CRDs (Application and AppProject), the configuration of Argo CD in a ConfigMap and the Argo CD secrets (SSH key and admin user).
```
Comment (tobru) [inline docs/modules/ROOT/pages/index.adoc:40]:
```suggestion
As soon as Steward could connect to the API and got the necessary information it starts to bootstrap Argo CD. The initial setup consists of the default deployments required to run Argo CD (`argocd-application-controller`, `argocd-redis, argocd-repo-server` and `argocd-server`), the Argo CD CRDs (`Application` and `AppProject`), the configuration of Argo CD in a ConfigMap and the Argo CD secrets (SSH key and admin user).
```

File: docs/modules/ROOT/pages/index.adoc [hunk 5]
```diff
@@ -1,6 +1,45 @@
...
+
+The SSH key pair (for Git) is generated on first and stored in a secret. The public key is sent to the API. The Argo CD admin user is configured with the Steward token as password to allow debugging of Argo CD via `kubectl port-forward`.
```
Comment (tobru) [inline docs/modules/ROOT/pages/index.adoc:42]:
```suggestion
The SSH key pair (for access to a Git repository via SSH) is generated on the first run of Steward and stored in a secret. The public key is sent to the API. The Argo CD admin user is configured with the Steward token as password to allow debugging of Argo CD via `kubectl port-forward`.
```

**Pull request merged**